spot-contracts/contracts/FeePolicy.sol

```diff
@@ -178,30 +190,62 @@ contract FeePolicy is IFeePolicy, OwnableUpgradeable {
-        require(p.lower >= -int256(SIGMOID_BOUND), "FeeStrategy: sigmoid lower bound too low");
-        require(p.upper <= int256(SIGMOID_BOUND), "FeeStrategy: sigmoid upper bound too high");
-        require(p.lower <= p.upper, "FeeStrategy: sigmoid asymptotes invalid");
-        perpRolloverFee.lower = p.lower;
-        perpRolloverFee.upper = p.upper;
-        perpRolloverFee.growth = p.growth;
+    /// @notice Updates the vault's deployment fee parameters.
+    /// @param vaultDeploymentFee_ The new deployment fee denominated in the underlying tokens.
+    function updateVaultDeploymentFees(uint256 vaultDeploymentFee_) external onlyOwner {
+        vaultDeploymentFee = vaultDeploymentFee_;
+    }
+
+    /// @notice Updates the swap fee percentage for a given pair of assets.
+    /// @dev Setting fees to 100% or 1.0 will effectively pause swapping.
+    /// @param assetIn The asset swapped in.
+    /// @param assetOut The vault asset swapped out.
+    /// @param feePerc The swap fee percentage.
+    function updateVaultSwapFees(
+        address assetIn,
+        address assetOut,
+        uint256 feePerc
+    ) external onlyOwner {
+        require(feePerc <= ONE, "FeeStrategy: perc too high");
+        bytes32 pairHash = keccak256(abi.encodePacked(assetIn, assetOut));
+        if (feePerc > 0) {
+            vaultSwapFeePerc[pairHash] = feePerc;
+        } else {
+            delete vaultSwapFeePerc[pairHash];
+        }
     }
 
     /// @inheritdoc IFeePolicy
-    function computePerpMintFeePerc() external override returns (uint256) {
-        return isOverSubscribed() ? 0 : perpMintFeePerc;
+    function computePerpMintFeePerc(uint256 perpValueIn) external override returns (uint256) {
+        // The act of minting more perps reduces the subscription ratio,
+        // We thus have to check if the "post"-minting subscription state is healthy and
+        // account for fees accordingly.
+        IFeePolicy.SubscriptionState memory postMintState = computeSubscriptionState(
+            perp.getDepositBond(),
+            perp.getTVL() + perpValueIn,
+            vault.getTVL()
+        );
+
+        // When the system is under-subscribed there exists an active mint fee
+        return (postMintState.normalizedSubscriptionRatio <= 1) ? perpMintFeePerc : 0;
     }
 
     /// @inheritdoc IFeePolicy
-    function computePerpBurnFeePerc() external override returns (uint256) {
-        return isOverSubscribed() ? perpBurnFeePerc : 0;
+    function computePerpBurnFeePerc(uint256 perpAmtBurnt, uint256 perpTotalSupply) external override returns (uint256) {
+        // The act of burning perps increases the subscription ratio,
+        // We thus have to check if the "post"-burning subscription state to account for fees.
+        IFeePolicy.SubscriptionState memory postBurnState = computeSubscriptionState(
+            perp.getDepositBond(),
+            perp.getTVL().mulDiv(perpTotalSupply - perpAmtBurnt, perpTotalSupply),
```

Comment:
I would have expected `perp.getTVL() - perpAmtBurnt`. Is there a reason for the mulDiv?

Reply:
TVL is denominated in the underlying asset. So we'd have to do `perp.getTVL() - (perpAmtBurnt*perp.getAvgPrice)` ..

Reply:
Ah makes sense. I'd add a comment, since that's important.

Reply:
Sure added.